nengo_loihi/builder/connection.py

```diff
@@ -740,7 +775,8 @@ def build_conv2d_connection(model, conn):
 
     gain = model.params[conn.post_obj.ensemble].gain
     if not np.all(gain == gain[0]):
-        # TODO: support this?
+        # Cannot fold gains into weights, result would not be convolutional.
+        # Therefore, Loihi does not support this if we want to share weights.
```

Comment:
Couldn't you use something like the decode interneurons to support this? I.e. the convolutional connection would target one set of compartments (with no gain), and then you'd have a second layer implementing the gains.

Reply:
We could definitely use that idea. But our decode-neuron system right now is designed for representing vectors in state-space, so there's multiple neurons per dimension. It'd take some modification.

Even then, there are considerable downsides to this. It changes the dynamics of the network, you need to figure out good spike rates that allow fast information transfer but don't affect the computation due to Loihi aliasing, and it doubles the number of neurons you need. It would be surprising to a user if all these changes happen just because they switch the gains from a homogeneous value to heterogeneous ones. It's also not needed for convnets, which are the main motivation for the Convolution transform.

Maybe at some point we could add an opt-in way to do this, but frankly I'm not sure if it'll ever happen, unless some pressing use-case develops that I'm not forseeing.

Reply:
Yeah that makes sense to me :+1: